homeassistant/components/xiaomi_miio/fan.py

```diff
@@ -1103,6 +1163,84 @@ async def async_set_dry_off(self):
         )
 
 
+class XiaomiAirHumidifierMiot(XiaomiAirHumidifier):
+    """Representation of a Xiaomi Air Humidifier (MiOT protocol)."""
+
+    @property
+    def speed(self):
+        """Return the current speed."""
+        if self._state:
+            if (
+                AirhumidifierMiotOperationMode(self._state_attrs[ATTR_MODE])
```

Comment:
Construct the operation mode container only once, and use a dict for mapping to simplify the code:
 and re-use that for checks to keep the code cleaner:
```
mode_mapping = {
  AirhumidifierMiotOperationMode.Mid: SPEED_LOW,
  AirhumidifierMiotOperationMode.Mid: SPEED_MEDIUM,
  AirhumidifierMiotOperationMode.High: SPEED_HIGH,
}

return mode_mapping.get(AirhumidiferMiotOperationMode(self._state_attrs[ATTR_MODE]))
```